test/integration/puppetlabs/services/master/environment_classes_int_test.clj

```diff
@@ -522,3 +550,123 @@
              "unexpected body for request with prior etag")
          (is (= expected-etag (response-etag response-with-tag))
              "unexpected etag returned for request with prior etag"))))))
+
+(defn create-jruby-instance-with-mock-class-info
+  [wait-atom class-info-atom]
+  (reify JRubyPuppet
+    (getClassInfoForEnvironment [_ _]
+      (let [class-info
+            {"/some/file"
+             (doto (HashMap.)
+               (.put
+                "classes"
+                @class-info-atom))}]
+        (when-let [promises @wait-atom]
+          (deliver (:wait-promise promises) true)
+          @(:continue-promise promises))
+        class-info))
+    (handleRequest [_ _]
+      (JRubyPuppetResponse. 0 nil nil nil))
+    (terminate [_]
+      (log/info "Terminating Master"))))
```

Comment:
![terminated](http://www.hdwallpapers.in/walls/t_800_terminator_genisys-wide.jpg)